### PR TITLE
Jena conversions and json parsing

### PR DESCRIPTION
**Pull request by bogdanromanx**

### CODE REVIEW

File: modules/jena/src/main/scala/ch/epfl/bluebrain/nexus/rdf/jena/syntax/FromJenaSyntax.scala [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package ch.epfl.bluebrain.nexus.rdf.jena.syntax
+
+import ch.epfl.bluebrain.nexus.rdf.{Graph, Node}
+import org.apache.jena.rdf.model
+import org.apache.jena.rdf.model.{Model, RDFNode, Resource}
+
+trait FromJenaSyntax {
+  import ch.epfl.bluebrain.nexus.rdf.jena.{JenaConverters => conv}
+
+  implicit class AsRdfModel(model: Model) {
```
Comment (wwajerowicz) [inline modules/jena/src/main/scala/ch/epfl/bluebrain/nexus/rdf/jena/syntax/FromJenaSyntax.scala]: I don't think I understand the logic behind the class names `AsRdfModel`. Wouldn't something like `AsRdfGraph` make more sense here?

File: modules/jena/src/main/scala/ch/epfl/bluebrain/nexus/rdf/jena/syntax/FromJenaSyntax.scala [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      * Converts a Jena [[org.apache.jena.rdf.model.Model]] to a [[ch.epfl.bluebrain.nexus.rdf.Graph]] using the
+      * argument `node` as an anchor.
+      *
+      * @param node the graph anchor
+      */
+    def asRdfGraph(node: Node): Either[String, Graph] =
+      conv.asRdfGraph(node, model)
+  }
+
+  implicit class AsRdfResource(resource: Resource) {
```
Comment (wwajerowicz) [inline modules/jena/src/main/scala/ch/epfl/bluebrain/nexus/rdf/jena/syntax/FromJenaSyntax.scala]: Similarly here, wouldn't `AsRdfIriOrBNode` make more sense here?

File: modules/jena/src/main/scala/ch/epfl/bluebrain/nexus/rdf/jena/ToJenaConverters.scala [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    */
+  def asJena(literal: Literal): model.Literal = literal match {
+    case Literal(lf, _, Some(LanguageTag(value))) => ResourceFactory.createLangLiteral(lf, value)
+    case Literal(lf, dt, _) =>
+      Try {
+        val tpe     = TypeMapper.getInstance().getSafeTypeByName(dt.asString)
+        val literal = ResourceFactory.createTypedLiteral(lf, tpe)
+        literal.getValue // It will throw whenever the literal does not match the desired datatype
+        literal
+      }.getOrElse(ResourceFactory.createStringLiteral(lf))
```
Comment (umbreak) [inline modules/jena/src/main/scala/ch/epfl/bluebrain/nexus/rdf/jena/ToJenaConverters.scala]: Are there tests for this? I'm not able to find them

File: modules/jena/src/test/scala/ch/epfl/bluebrain/nexus/rdf/jena/Fixture.scala [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+import ch.epfl.bluebrain.nexus.rdf.{Graph, Iri}
+import ch.epfl.bluebrain.nexus.rdf.syntax.all._
+import ch.epfl.bluebrain.nexus.rdf.Node.Literal
+import ch.epfl.bluebrain.nexus.rdf.Node.Literal.LanguageTag
+import ch.epfl.bluebrain.nexus.rdf.Vocabulary.{rdf, schema, xml}
+import org.apache.jena.rdf.model.{AnonId, Model, ModelFactory, Resource, ResourceFactory}
+import org.apache.jena.rdf.model.impl.ResourceImpl
+
+object Fixture {
```
Comment (umbreak) [inline modules/jena/src/test/scala/ch/epfl/bluebrain/nexus/rdf/jena/Fixture.scala]: Are there tests for default Jena mapper values? like `schema:Date`, `xsd:Long`, etc:
```scala
      val jenaLiteral =Literal("1999-04-09T20:00Z", url"http://schema.org/Date".value). asJena
      jenaLiteral.getLexicalForm shouldEqual "1999-04-09T20:00Z"
      jenaLiteral.getDatatypeURI shouldEqual "http://schema.org/Date"
```

File: modules/jena/src/main/scala/ch/epfl/bluebrain/nexus/rdf/jena/syntax/FromJenaSyntax.scala (hunk 1 above)
Comment (bogdanromanx) [reply modules/jena/src/main/scala/ch/epfl/bluebrain/nexus/rdf/jena/syntax/FromJenaSyntax.scala]: There are naming collisions if we change the naming based on the target. Maybe inverting would be better: `ModelAsRdf`?

Comment (wwajerowicz) [reply modules/jena/src/main/scala/ch/epfl/bluebrain/nexus/rdf/jena/syntax/FromJenaSyntax.scala]: that sounds better.

File: modules/jena/src/main/scala/ch/epfl/bluebrain/nexus/rdf/jena/ToJenaConverters.scala (hunk 3 above)
Comment (bogdanromanx) [reply modules/jena/src/main/scala/ch/epfl/bluebrain/nexus/rdf/jena/ToJenaConverters.scala]: These are tested through the syntax to cover both the syntax and the logic:
```
"node is a literal" in {
  val cases = List(
    ResourceFactory.createLangLiteral("a", "en-US")                 -> Literal("a", LanguageTag("en-US").rightValue),
    ResourceFactory.createStringLiteral("a")                        -> Literal("a"),
    ResourceFactory.createTypedLiteral("1", XSDDatatype.XSDinteger) -> Literal(1),
    ResourceFactory.createTypedLiteral("1.0", XSDDatatype.XSDfloat) -> Literal(1f),
    ResourceFactory.createLangLiteral("1", "en")                    -> Literal("1", LanguageTag("en").rightValue)
  )
  forAll(cases) {
    case (jl, l) =>
      (l: Node).asJena shouldEqual jl
      l.asJena shouldEqual jl
  }
}
```

File: modules/jena/src/test/scala/ch/epfl/bluebrain/nexus/rdf/jena/Fixture.scala (hunk 4 above)
Comment (bogdanromanx) [reply modules/jena/src/test/scala/ch/epfl/bluebrain/nexus/rdf/jena/Fixture.scala]: there are some tests that assert that the lexical form and the datatype are propagated properly (see above comment). The example in your comment tests this exact thing (datatype and lexicalform are propagated correctly).

**Pull request merged**